src/core/CLucene/index/IndexWriter.cpp

```diff
@@ -1879,7 +1884,8 @@ void IndexWriter::mergeTerms(bool hasProx, IndexVersion indexVersion) {
             TermInfo termInfo;
             termInfo.set(dfs[i], freqPointer, proxPointer, (int32_t) (skipPointer - freqPointer));
             // Write a new TermInfo
-            termInfosWriter->add(smallestTerm, &termInfo);
+            std::string cur_term = lucene_wcstoutf8string(smallestTerm->text(), smallestTerm->textLength());
+            termInfosWriter->add(smallestTerm->field(), cur_term.data(), cur_term.length(), &termInfo);
```

Comment:
What's the effect of the change here and why change it?

Reply:
I dig into your newly added function add() but can not find the difference.

Reply:
Because we have two types in clucene: STermInfosWriter and TermInfosWriter, I only modified the code logic in STermInfosWriter. My expectation is to maintain only one process within STermInfosWriter.

Reply:
STerm<T>